### DISCUSSION TITLE
在macOS上用VSCode写C代码

### DISCUSSION THREAD

**Discussion by Sinvee0102** [Yang-Xijie/yang-xijie.github.io#7] [Cpp in VSCode on macOS]
教程中只有关于C++的，想知道写C代码有没有类似的方法，谢谢

**Comment by Yang-Xijie** [comment]
`C++`是`C`的超集。也就是说，你在`.cpp`文件里面写C代码是完全没有问题的。

如果你一定要写纯C代码，那就把教程里面所有g++都换成gcc，把所有c++都换成c就行了。配置的思路是完全一样的

**Reply by Shizhuo-Zhu** [reply]
hello 你好，我想请问下在vs code里用auto定义变量时总会报错是什么情况？

<img width="774" alt="Screen Shot 2021-11-12 at 18 05 09" src="https://user-images.githubusercontent.com/93555595/141601850-270e9c6d-dfa6-4b31-a98b-b58c17dc8fad.png">

**Reply by Yang-Xijie** [reply]
我没怎么用过`auto`，如果你确定你的用法没问题，可以检查一下`C/C++`插件的版本是不是最新的。
不然就是VS Code抽了…

**Reply by Je2ry** [reply]
请问你的问题解决了吗，我目前也遇到这种问题，c++11的语法都可以编译运行，但是总是报错提示

**Comment by Cycle02** [comment]
up主 我在第7课调试时 mac的系统终端出现了这个 请问该怎么解决 谢谢
--disable-aslr] [--no-env] [--setsid] [--help] -- <PROGRAM> [<PROGRAM-ARG> <PROGRAM-ARG> ....]

DESCRIPTION
    darwin-debug will exec itself into a child process <PROGRAM> that is
    halted for debugging. It does this by using posix_spawn() along with
    darwin specific posix_spawn flags that allows exec only (no fork), and
    stop at the program entry point. Any program arguments <PROGRAM-ARG> are
    passed on to the exec as the arguments for the new process. The current
    environment will be passed to the new process unless the "--no-env"
    option is used. A unix socket must be supplied using the
    --unix-socket=<SOCKET> option so the calling program can handshake with
    this process and get its process id.

EXAMPLE
   darwin-debug --arch=i386 -- /bin/ls -al /tmp

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]

**Reply by Yang-Xijie** [reply]
啊 我不太清楚这是什么问题
想问一下你调试小demo也会出这样的报错吗？
还有就是，如果你用的Mac是M1的Mac的话，教程中的调试步骤应该会出问题。怎么解决我目前还不清楚
你可以再提供多一些的报错信息，比如这段报错之前还有什么样的信息？
推荐先用google或者百度查一下，也许会知道问题在哪里
（另外你的问题和这个discussion不太一样，之后提不同的问题可以重新开一个discussion哦）

**Comment by Shizhuo-Zhu** [comment]
还想请教下up主，VS code在初始化vector时不能用{}而得用[]吗？我每次用花括号定义就报错但是可以运行。

**Reply by Yang-Xijie** [reply]
首先，C/C++语言的语法是根据C/C++语言的标准确定的，VS Code只是编辑器，报错是通过插件或者是编译器实现的。我建议你参考「报错」信息中的警告(warning)或是错误(error)来对代码进行修改，必要时可能要去查官方的文档看vector到底怎么应该定义。

**Comment by Shizhuo-Zhu** [comment]
请问下up主如何同时编译多个.cpp文件呢？

**Reply by Yang-Xijie** [reply]
在脚本中直接添加需要编译的.cpp文件就可以了，有多少加多少
我在视频中应该说过

**Reply by Yang-Xijie** [reply]
https://yang-xijie.github.io/LECTURE/vscode_cpp_macos/10_极速版.html

**Reply by Yang-Xijie** [reply]
settings.json 和 tasks.json 文件的注释里面我已经写的很清楚了

**Reply by Yang-Xijie** [reply]
另外，有新的问题新开一个discussion吧

**Reply by gitony0101** [reply]
> [https://yang-xijie.github.io/LECTURE/vscode_cpp_macos/10_极速版.html](https://yang-xijie.github.io/LECTURE/vscode_cpp_macos/10_%E6%9E%81%E9%80%9F%E7%89%88.html)

看不到啦
